diff -ru qtwayland-opensource-src-5.0.0-beta1/src/plugins/platforms/wayland/wayland.pro qtwayland-opensource-src-5.0.0-beta1-fixbuild/src/plugins/platforms/wayland/wayland.pro
--- qtwayland-opensource-src-5.0.0-beta1/src/plugins/platforms/wayland/wayland.pro	2012-08-28 21:41:35.000000000 +0200
+++ qtwayland-opensource-src-5.0.0-beta1-fixbuild/src/plugins/platforms/wayland/wayland.pro	2012-09-22 21:47:16.688525150 +0200
@@ -4,7 +4,6 @@
 
 CONFIG += link_pkgconfig qpa/genericunixfontdatabase
 
-DESTDIR = $$QT.compositor.plugins/platforms
 
 QT += core-private gui-private platformsupport-private
 
