diff -ru qt5-qtwayland/src/plugins/platforms/wayland/wayland.pro qt5-qtwayland-glib//src/plugins/platforms/wayland/wayland.pro
--- qt5-qtwayland/src/plugins/platforms/wayland/wayland.pro	2012-06-18 21:52:07.000000000 +0200
+++ qt5-qtwayland-glib//src/plugins/platforms/wayland/wayland.pro	2012-06-23 12:14:02.819660201 +0200
@@ -82,7 +82,7 @@
 INCLUDEPATH += $$PWD/../../../shared
 
 !contains(QT_CONFIG, no-pkg-config) {
-    PKGCONFIG += wayland-client
+    PKGCONFIG += wayland-client glib-2.0
 } else {
     LIBS += -lwayland-client
 }
Only in qt5-qtwayland-glib//src/plugins/platforms/wayland: wayland.pro~
